pkg/controller/backup_repository_controller.go

```diff
@@ -194,8 +204,14 @@ func (r *BackupRepoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	bsl, bslErr := r.getBSL(ctx, backupRepo)
+	if bslErr != nil {
+		log.WithError(bslErr).Error("Fail to update BackupRepository status, when BSL is not found for that BackupRepository.")
+		return ctrl.Result{}, nil
```

Comment:
Is there any difference of returning `nil` or error under this case?

Reply:
return error will cause the same request to retry in the reconcile.
It seems retry on the BackupRepositories without BSL has no meaning.